src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSchemaObject.java

```diff
@@ -386,6 +390,58 @@ public static CreateCollectionCommand collectionSettingToCreateCollectionCommand
 
     // CreateCollectionCommand object is created for convenience to generate json
     // response. The code is not creating a collection here.
-    return new CreateCollectionCommand(collectionSetting.collectionName(), options);
+    return new CreateCollectionCommand(collectionSetting.name.table(), options);
+  }
+
+  public IdConfig idConfig() {
+    return idConfig;
+  }
+
+  public IndexingConfig indexingConfig() {
+    return indexingConfig;
+  }
+
+  // TODO: these helper functions break encapsulation for very little benefit
+  public CollectionSchemaObject.SimilarityFunction similarityFunction() {
+    return vectorConfig().similarityFunction();
+  }
+
+  public boolean isVectorEnabled() {
+    return vectorConfig() != null && vectorConfig().vectorEnabled();
+  }
+
+  // TODO: the overrides below were auto added when migrating from a record to a class, not sure
+  // they are needed or wanted
```

Comment:
Had the same question: guessing they really shouldn't be needed TBH.

Reply:
I don't think these overrides are used anywhere

Reply:
Will check removal as follow-up, not with initial refactoring.